plugins/metrics/plugin.go

```diff
@@ -166,6 +168,46 @@ func registerLocalMetrics() {
 		increasePerComponentCounter(Booker)
 	}))
 
+	messagelayer.Tangle().ApprovalWeightManager.Events.MessageFinalized.Attach(events.NewClosure(func(messageID tangle.MessageID) {
+		messageType := DataMessage
+		messagelayer.Tangle().Utils.ComputeIfTransaction(messageID, func(_ ledgerstate.TransactionID) {
+			messageType = Transaction
+		})
+
+		var issuingTime time.Time
+		messagelayer.Tangle().Storage.Message(messageID).Consume(func(message *tangle.Message) {
+			message.ForEachParent(func(parent tangle.Parent) {
+				increasePerParentType(parent.Type)
+			})
+			issuingTime = message.IssuingTime()
+		})
+		if messagelayer.Tangle().Storage.MessageMetadata(messageID).Consume(func(messageMetadata *tangle.MessageMetadata) {
+			messageFinalizationTotalTime[messageType] += uint64(messageMetadata.FinalizedTime().Sub(issuingTime).Milliseconds())
```

Comment:
this should be done in nanoseconds

Reply:
Nanoseconds could overflow for uint64 and for metrics we don't need that much precision anyway.